--- configure.in~	2004-11-14 20:55:15.115487624 +0000
+++ configure.in	2004-11-14 20:59:35.805856664 +0000
@@ -231,6 +231,15 @@
         HostVendor_CPP='unknown'
         HostOS_CPP='linux'
         ;;
+x86_64-*-linux*)
+        HostPlatform=x86_64-unknown-linux # hack again
+        TargetPlatform=x86_64-unknown-linux
+        BuildPlatform=x86_64-unknown-linux
+        HostPlatform_CPP='x86_64_unknown_linux'
+        HostArch_CPP='x86_64'
+        HostVendor_CPP='unknown'
+        HostOS_CPP='linux'
+        ;;
 m68k-*-linux*)
         HostPlatform=m68k-unknown-linux # hack again
         TargetPlatform=m68k-unknown-linux
